Remove nvConvexDecomp and replace it with v-hacd

marauder2k7 opened the pull request:
Updated the lib used for convex hull generation.

note: needs tested on other platforms.

marauder2k7 committed:
vhacd

added vhacd not working atm

marauder2k7 committed:
working vhacd

renamed ThreadPool to TorqueThreadPool to avoid conflics
fixed data transmission between stages of convexDecome and trimesh creation
TODO: re-add our own functions for generating sphere/cylinder/box

marauder2k7 committed:
final cleanup

final cleanup removal of the old convexDecomp library
ADDDED: library for the floatmath from v-hacd resource, required for fit sphere/capsule/box functions

marauder2k7 committed:
Update tsMeshFit.cpp

code cleanup, fix warnings etc

marauder2k7 committed:
linux and tests

fix linux build
change thread test to use TorqueThreadPool

marauder2k7 committed:
Update tsMeshFit.cpp

fix addSphere
fix addCapsule

now spheres and capsules scale correctly.... hopefully....

dottools commented:
Why are there unrelated changes in this PR? Such as `ThreadPool` to `TorqueThreadPool`?

marauder2k7 commented:
> Why are there unrelated changes in this PR? Such as `ThreadPool` to `TorqueThreadPool`?

naming conflict with the resource as it also has a class called threadpool

marauder2k7 committed:
revert ThreadPool rename

revert ThreadPool rename, resources ThreadPool class is already nested in namespace VHACD

marauder2k7 commented:
> Why are there unrelated changes in this PR? Such as `ThreadPool` to `TorqueThreadPool`?

turns out the ThreadPool class in the resource is nested in a namespace so we should be fine......... reverted rename back
